fiftyone/core/odm/database.py

```diff
@@ -173,15 +179,58 @@ def _connect():
     global _client
     if _client is None:
         global _connection_kwargs
-        _client = pymongo.MongoClient(**_connection_kwargs)
+        _client = pymongo.MongoClient(
+            **_connection_kwargs, appname=foc.DATABASE_APPNAME
+        )
         connect(fo.config.database_name, **_connection_kwargs)
 
 
 def _async_connect():
     global _async_client
     if _async_client is None:
         global _connection_kwargs
-        _async_client = motor.motor_tornado.MotorClient(**_connection_kwargs)
+        _async_client = motor.motor_tornado.MotorClient(
+            **_connection_kwargs, appname=foc.DATABASE_APPNAME
+        )
+
+
+def _delete_non_persistent_datasets(config, client):
+    db = client[config.database_name]
+
+    has_datasets = False
+    for doc in db.datasets.find(
+        {"persistent": False},
+        {
```

Comment:
Side comment: I appreciate the optimization here, but FYI the dataset doc is intended to always be quite small, so I wouldn't bat an eye if you just used `db.datasets.find({"persistent": False})`

Reply:
oh yeah that's fine, I was just saying you could use `for doc in db.datasets.find({"persistent": False})` in this method 😉

Reply:
Oh I see. My mistake, that makes total sense :+1:

Reply:
@brimoor Aren't there grid fs files associated with a dataset that need to be cleaned up? Should [this function](https://github.com/voxel51/fiftyone/blob/cf0a75d9ec5352db8027a105bcd300c54fab0d53/fiftyone/core/dataset.py#L176) be used?

Reply:
That's the one I thought @brimoor was talking about. See my first comment in this thread about circular dependencies.

Reply:
Ok, I'll take a look

Reply:
Good catch @benjaminpkane: yes, `Dataset.delete()` really does need to be called.

@j053y `fiftyone.core.odm.database.delete_dataset()` isn't intended to be a "public" method, more of a last resort utility to (mostly) get a corrupted dataset out of the user's deployment

Reply:
@brimoor I don't see anyway to use `Dataset.delete()` with major refactoring due to the circular import problem. Also I'm not currently using `fiftyone.core.odm.database.delete_dataset()`

Reply:
I brought up `fiftyone.core.odm.database.delete_dataset()` just to clarify that its logic is equivalent to your logic in `_autodelete_non_persistent_datasets()`, but that's okay for that method because it is designed to be an internal-only thing.

However, `_autodelete_non_persistent_datasets()` needs to do a full deletion like `Dataset.delete()` does or else users will accumulate orphan content in the DB.